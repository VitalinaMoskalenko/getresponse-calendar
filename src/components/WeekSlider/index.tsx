import React from "react";
import styled from "styled-components";
import Button from "../Button";
import { H3 } from "../Headings";
import WeekItem from "../WeekItem";

const Container = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  margin: 0px;
`;

type PropsType = {
  countWeek: number;
};

const WeekSlider = ({ countWeek }: PropsType) => {
  const goToAnotherWeek = (i: number) => {
    console.log(i);
  };

  return (
    <Container>
      {[...Array(countWeek)].map((e, i) => (
        <WeekItem
          key={i}
          numberWeek={i + 1}
          countWeek={countWeek}
          onClick={() => goToAnotherWeek(i + 1)}
        />
      ))}
    </Container>
  );
};

export default WeekSlider;
