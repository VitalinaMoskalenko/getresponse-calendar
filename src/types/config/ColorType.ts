export enum ColorType {
  white = "white",
  black = "black",
  lightGrey = "#e1e1e1",
  darkGrey = "#535353",
  green = "#b2c200",
  orange = "#ff801a",
}
