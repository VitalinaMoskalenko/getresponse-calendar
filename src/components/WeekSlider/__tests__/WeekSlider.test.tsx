import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import WeekSlider from "../";
import { theme } from "../../../config";

it("renders WeekSlider correctly", () => {
  const weekSliderComponent = renderer
    .create(
      <ThemeProvider theme={theme}>
        <WeekSlider onValueChanged={() => {}} countWeek={1} numberWeek={1} />
      </ThemeProvider>
    )
    .toJSON();

  expect(weekSliderComponent).toMatchSnapshot();
});
