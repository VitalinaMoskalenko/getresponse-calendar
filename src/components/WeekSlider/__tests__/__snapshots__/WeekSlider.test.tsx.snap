// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders WeekSlider correctly 1`] = `
<div
  className="sc-kEqXSa ljIMGs"
>
  <h1
    className="sc-fujyAs sc-crzoAE EUgGk keCRtU"
  >
    Week 
    1
  </h1>
</div>
`;
